lib/analysis/network.py

```diff
@@ -45,26 +44,27 @@ def message_number_graph(log_dict, nicks, nick_same_list, DAY_BY_DAY_ANALYSIS =
                     corrected_nick = util.correctLastCharCR(parsed_nick.group(0)[1:-1])
                     nick_sender = ""
                     nick_receiver = ""
-                    for i in range(config.MAX_EXPECTED_DIFF_NICKS):
-                        if ((i < len(conn_comp_list)) and (corrected_nick in conn_comp_list[i])):  #change nick_same_list to conn_comp_list because conn_comp_list is the main list of all users and nicks now
-                            nick_sender = conn_comp_list[i][0]
-                            break
+                    nick_sender = nick_same_list_to_conn_comp_list(conn_comp_list, corrected_nick)        
 
                     for nick in nicks:
                         rec_list = [e.strip() for e in line.split(':')]
                         util.rec_list_splice(rec_list)
                         if not rec_list[1]:
                             break
-                        for k in xrange(len(rec_list)):
-                            if(rec_list[k]):
-                                rec_list[k] = util.correctLastCharCR(rec_list[k])
+                        #for k in xrange(len(rec_list)):
+                            #if(rec_list[k]):
+                                #rec_list[k] = util.correctLastCharCR(rec_list[k])
+                        rec_list = util.correct_last_char_list(rec_list)        
                         for receiver in rec_list:
                             if(receiver == nick):
                                 if(corrected_nick != nick):
-                                    for i in range(config.MAX_EXPECTED_DIFF_NICKS):
-                                        if ((i<len(conn_comp_list)) and (nick in conn_comp_list[i])):
-                                            nick_receiver=conn_comp_list[i][0]
-                                            break
+                                    #for i in range(config.MAX_EXPECTED_DIFF_NICKS):
+                                        #if ((i<len(conn_comp_list)) and (nick in conn_comp_list[i])):
+                                            #nick_receiver=conn_comp_list[i][0]
+                                            #break
+                                    nick_receiver = nick_same_list_to_conn_comp_list(conn_comp_list, nick)    
```

Comment:
I'm not really sure if this function will work correctly here, since the break might not be executed.
Please confirm this.
Also, check outputs of some other graphs to confirm that you aren't breaking any code.